keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -21,22 +19,19 @@ class ArtifactListener(
   fun onArtifactEvent(event: ArtifactEvent) {
     log.info("Received artifact event: {}", event)
     event.artifacts.forEach {
-      DeliveryArtifactVersion(
-        DeliveryArtifact(it.name, ArtifactType.valueOf(it.type)),
-        it.version,
-        it.provenance.let(URI::create)
-      )
-        .apply {
-          if (artifactRepository.isRegistered(artifact.name, artifact.type)) {
-            log.info("Registering version {} of {} {}", version, artifact.name, artifact.type)
-            artifactRepository.store(this)
-              .also { wasAdded ->
-                if (wasAdded) {
-                  publisher.publishEvent(ArtifactVersionUpdated(artifact.name, artifact.type))
-                }
-              }
+      val artifact = DeliveryArtifact(it.name, ArtifactType.valueOf(it.type))
+      // TODO: should be able to construct this with Frigga or something, apparently, also it might
```

Comment:
I think you can only parse with Frigga, but I might be mistaken.

Reply:
I'll have a dig. I don't think I currently have Frigga on the classpath.